From fd0e6557278691ca02eda1f505ee35b2455822e1 Mon Sep 17 00:00:00 2001
From: Andy Ross <andy.ross@windriver.com>
Date: Thu, 2 May 2013 11:21:20 -0700
Subject: Fix BufferQueue drain timeout

Commit 3df059d4 ("Fix hang under BufferQueue::disconnect") had a bad
value.  The unit is nanoseconds, not microseconds.  The mistaken 300us
timeout almost worked, but would result in spurious BufferQueue
failures in conditions (e.g. CTS runs) of heavy scheduler load.

Change-Id: I308ddba1f82b5d60be1318d9c37b75a7e8051b14
For: AXIA-2552
Signed-off-by: Andy Ross <andy.ross@windriver.com>
---
 libs/gui/BufferQueue.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libs/gui/BufferQueue.cpp b/libs/gui/BufferQueue.cpp
index 1a867fc..12e8289 100644
--- a/libs/gui/BufferQueue.cpp
+++ b/libs/gui/BufferQueue.cpp
@@ -1000,7 +1000,7 @@ void BufferQueue::freeAllBuffersExceptHeadLocked() {
 }
 
 status_t BufferQueue::drainQueueLocked() {
-    const nsecs_t timeout = 300000; // 0.3s seems a reasonable wait for a valid consumer
+    const nsecs_t timeout = 300000000; // 0.3s seems a reasonable wait for a valid consumer
     nsecs_t start = systemTime();
     while (mSynchronousMode && mQueue.size() > 1) {
         mDequeueCondition.waitRelative(mMutex, timeout);
-- 
1.7.4.1

